site/layouts/partials/head.html

```diff
@@ -2,12 +2,32 @@
 <meta http-equiv="X-UA-Compatible" content="IE=edge">
 <title> {{ .Title }}</title>
 <meta name="viewport" content="width=device-width, initial-scale=1"> {{ with .Site.Params.author }}
-<meta name="author" content="{{ . }}">{{ end }} {{ with .Site.Params.description }}
-<meta name="description" content="{{ . }}">{{ end }} {{ with .Site.LanguageCode }}
+<meta name="author" content="{{ . }}">{{ end }} {{ with .Site.LanguageCode }}
 <meta http-equiv="content-language" content="{{ . }}" />{{ end }}
 
-<!-- canonical link tag -->
-{{ if .Params.canonicalUrl }}
+<!-- Meta Descriptions --> {{ if .Description }}
+  {{ partial "head-meta.html" .Description }}
+{{ else }}
+  {{ if .IsSection }}
+    {{ range where .Site.Params.indexes.item "relPermaLink" "==" $.RelPermalink }}
+      {{ partial "head-meta.html" .description }}
+    {{ else }}
+      {{ partial "head-meta.html" .Site.Params.Description }}
+    {{ end }}
+  {{ else }}
+    {{ if eq .Type "categories" }}
+      {{ range where .Site.Params.categories.item "term" "==" $.Data.Term }}
+        {{ partial "head-meta.html" .description }}
+      {{ else }}
+        {{ partial "head-meta.html" .Site.Params.categories.description }}
+      {{ end }}
+    {{ else }}
+      {{ partial "head-meta.html" .Site.Params.Description }}
+    {{ end }}
+  {{ end }}
+{{ end }}
```

Comment:
Meta description values do not always come from the same place. This set of conditions will find the meta description based on whether or not it has a description defined in `.md` file, or if it should fall back to settings that come from `config.toml` or if none is found at all, falling back to the site default meta description.

Reply:
Is that additional logic really needed? Is it simple to just add an `_index.md` with the metadata?

Reply:
I'm really not tasked with making content updates, so I didn't want to alter the structure of how that content is populated. Since it's compiled prior to going to production, I didn't see the issue with this approach. On the one hand, it allows the existing content structure in terms of markdown files to remain the same; on the other, editors would need to go the config file to make updates for those pages that do not have index files (I'm guessing it's not that often since they don't have markdown files already).

Reply:
I'm unsure of whether or not that's possible with taxonomy (category/tag) pages.

Reply:
Checking in for direction on which course to take.

Reply:
@lawrencejohnson I'm also unsure if my suggestion would work for taxonomies.

I'm fine with keeping the the code as it is for now and opening a help wanted ticket for a community member to pick up that deals with moving the descriptions to the md files.

Reply:
@hanzei @lawrencejohnson - Has a help wanted ticket been opened for this work?

Reply:
@cwarnermm - I'm happy to address if a ticket gets opened. My preference would be for specific direction on the desired approach. If the desired approach is to create `_index.md` files, then the content updates should happen before I strip down the code. My only note is that I'm fairly certain this can only be done for pages, but not taxonomies like categories. In the end, I don't anticipate more than one of these if blocks and its corresponding item search going away.

Reply:
Thanks @lawrencejohnson. @hanzei - are you able to review and provide specific direction on the desired approach? If not, I'll identify the correct reviewer for this.

Reply:
@lawrencejohnson I agree that for `taxonomies` we need to put the content into the config file. But for page bundles like `/blog/` we can use an `_index` instead. Do you think that is reasonable?

Reply:
@hanzei, it makes no difference to me really. Anyone can create the `_index` files at anytime and it should take precedence over the fallback in this conditional code. Once they are in place I'd be happy to edit out the check against `config.toml` for them. We may want to wait on that effort until the linked PR below goes through, though. There is another update coming through for a heading property that would probably be easier to transfer to index files at the same time.

https://github.com/mattermost/mattermost-developer-documentation/pull/764